src/terragraph-e2e/e2e/minion/BgpUtils.cpp

```diff
@@ -377,83 +365,45 @@ BgpUtils::runVtyshCmd(const std::string& command) {
-    std::vector<std::string>& entries) {
-  // Split output line by line
-  std::vector<folly::StringPiece> bgpTableLines;
-  folly::split('\n', bgpTableOutput, bgpTableLines);
-
-  int rowCount = 0;
-  bool inTable = false;
-
-  for (const auto& line : bgpTableLines) {
-    // Split line by whitespace
-    std::string lineStr = line.toString();
-    std::istringstream lineStrStream(lineStr);
-    std::vector<std::string> lineEntries{
-        std::istream_iterator<std::string>{lineStrStream},
-        std::istream_iterator<std::string>{}};
-
-    if (inTable) {
-      // Table is finished when the next line is empty
-      if (lineStr.empty()) {
-        inTable = false;
-        continue;
-      }
+// Vtysh Bgp Advertised Routes Info List
+std::vector<thrift::BgpRouteInfo>
+BgpUtils::createVtyshBgpAdvertisedRouteInfoList(const std::string &advertisedRoutes, const std::string &key, const std::string &value) {
 
-      entries.insert(entries.end(), lineEntries.begin(), lineEntries.end());
-    } else {
-      // Not in the table
-      if (!lineEntries.empty() && lineEntries[0] == firstHeader) {
-        // We are at the start of the table
-        headers.insert(headers.end(), lineEntries.begin(), lineEntries.end());
-        inTable = true;
-      } else if (line.startsWith("Total number")) {
-        // We are at the section that states the number of neighbors/prefixes
-        rowCount = folly::to<int>(lineEntries.back());
-        break;
-      } else if (line.startsWith("Displayed") && lineEntries.size() == 7) {
-        rowCount = folly::to<int>(lineEntries[1]);
-        break;
-      }
+    // Network and Next Hop
+    std::vector<thrift::BgpRouteInfo> routeInfo;
+    folly::dynamic advertisedRoutesInfo = folly::dynamic::object;
+    advertisedRoutesInfo = folly::parseJson(advertisedRoutes)[key.c_str()];
+    thrift::BgpRouteInfo bgpRouteInfo;
+    for (const auto &routes : advertisedRoutesInfo.items()) {
+        // Advertised Routes Information
+        std::string network = routes.first.asString();
+        bgpRouteInfo.network = network;
+        bgpRouteInfo.nextHop = advertisedRoutesInfo[network.c_str()][value.c_str()].asString();
+        routeInfo.push_back(bgpRouteInfo);
     }
-  }
 
-  return rowCount;
+    return routeInfo;
 }
 
+// Vtysh Bgp Recived Routes Info List
 std::vector<thrift::BgpRouteInfo>
-BgpUtils::createVtyshBgpRouteInfoList(
-    int rowCount, const std::vector<std::string>& entries) {
-  // Since each row can have a variable number of entries
-  // (due to the variable path length), if the next entry is a 'i', 'e' or '?'
-  // then it marks the end of the row.
-  std::vector<thrift::BgpRouteInfo> routeInfo;
-  unsigned int rowStartIdx = 0;
-  for (int i = 0; i < rowCount; ++i) {
-    // Skip the first entry on each row since it is the status code
+BgpUtils::createVtyshBgpRecivedRouteInfoList(const std::string &recivedRoutes, const std::string &key, const std::string &value) {
```

Comment:
- fix typo "Recived"
- 80 chars per line

Reply:
Addressed the review comments